packages/env/lib/wordpress.js

```diff
@@ -27,6 +27,46 @@ const { getCache, setCache } = require( './cache' );
  * @typedef {'development'|'tests'|'all'} WPEnvironmentSelection
  */
 
+/**
+ * Utility function to check if a version is lower than another version, only considering the major and minor portions.
+ *
+ * This is a non-comprehensive check only intended for this usage, to avoid pulling in a full semver library.
+ *
+ * @param {string} version        The version to check.
+ * @param {string} compareVersion The compare version to check whether the version is lower than.
+ * @return {boolean} True if the version is lower than the compare version, false otherwise.
+ */
+function isMajorMinorVersionLower( version, compareVersion ) {
```

Comment:
The name is a bit confusing, as it's not immediately clear which version is supposed to be lower one, plus in WP speak "6.5" is a major release anyway.

Also, the function is really hard to understand with the string splitting etc. Since this is not a hot code path and the string is very simple, this can be greatly simplified with a regex.

```js
function isMajorMinorVersionLower( version, compareVersion ) {
	const left = Number.parseFloat( version.match( /[0-9]+\.[0-9]+/ )[0] );
	const right = Number.parseFloat( compareVersion.match( /[0-9]+\.[0-9]+/ )[0] );

	return left < right;
}
```

Reply:
Agree to all you're saying, but not sure I like this better. It's shorter, but would produce an incorrect result on something like `5.9 < 5.10`.

Since it's not reused, why is it a problem that the code is more verbose if it's also more accurate? What the function is supposed to do is perform a version comparison that only considers the major and minor portions of the version. I think the current code does that correctly, while your proposed code doesn't per my example.

Reply:
WordPress will never have versions like 5.10, so supporting that case is a non-goal.

Both the cyclomatic and cognitive complexity of the current function are very high, especially without tests to consult. This 3-liner is much easier to grasp even without any tests and works under the constraints of WordPress' versioning system.

Reply:
That's fair. I'll use your approach and rename the function to `isWPMajorMinorVersionLower`. I think that should help clarify. Yes, it's a bulky name, but that doesn't really matter IMO since it's not exported / not a public API function.